src/fs.ts

```diff
@@ -0,0 +1,258 @@
+// This file is intended to be used only in a Node.js context.
+/* eslint-disable import/no-nodejs-modules */
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+import { isErrorWithCode, wrapError } from './errors';
+import type { Json } from './json';
+
+/**
+ * Information about the file sandbox provided to tests that need temporary
+ * access to the filesystem.
+ */
+export type FileSandbox = {
+  directoryPath: string;
+  withinSandbox: (
+    test: (args: { directoryPath: string }) => Promise<void>,
+  ) => Promise<void>;
+};
+
+/**
+ * Reads the file at the given path, assuming its content is encoded as UTF-8.
+ *
+ * @param filePath - The path to the file.
+ * @returns The content of the file.
+ * @throws An error with a stack trace if reading fails in any way.
+ */
+export async function readFile(filePath: string): Promise<string> {
+  try {
+    return await fs.promises.readFile(filePath, 'utf8');
+  } catch (error) {
+    throw wrapError(error, `Could not read file '${filePath}'`);
+  }
+}
+
+/**
+ * Writes content to the file at the given path.
+ *
+ * @param filePath - The path to the file.
+ * @param content - The new content of the file.
+ * @throws An error with a stack trace if writing fails in any way.
+ */
+export async function writeFile(
+  filePath: string,
+  content: string,
+): Promise<void> {
+  try {
+    await fs.promises.mkdir(path.dirname(filePath), { recursive: true });
```

Comment:
Maybe explain in the JSDoc that this function attempts to create the directory.

Reply:
Makes sense. Added in [3a5953e](https://github.com/MetaMask/utils/pull/148/commits/3a5953e9b1e51545292eb0a99a175da63b2a5043).